examples/customOpenAPIschema.md

```diff
@@ -0,0 +1,230 @@
+# Using a Custom OpenAPI schema
+
+A kustomization file supports adding your own
+OpenAPI schema to define merge keys and patch
+strategy.
+
+Make a place to work:
+
+<!-- @placeToWork @testAgainstLatestRelease -->
+```
+DEMO_HOME=$(mktemp -d)
+```
+
+We'll be editing our own CRD:
+
+<!-- @customOpenAPI @testAgainstLatestRelease -->
+```
+cat <<EOF >$DEMO_HOME/mycrd.yaml
+apiVersion: example.com/v1alpha1
+kind: MyCRD
+metadata:
+  name: service
+spec:
+  template:
+    spec:
+      containers:
+      - name: server
+        image: server
+        command: example
+        ports:
+        - name: grpc
+          protocol: TCP
+          containerPort: 8080
+EOF
+```
+
+The edit we want to make is change the image to nginx. We can
+do that via a strategic merge patch. 
+
+Kustomize gets its merge key information from the OpenAPI data
+provided by the kubernetes API server. It doesn't have information
+about custom resources, so we will have to provide our own 
+schema file.
```

Comment:
It would be instructive to add a note as to where this came from, or 
maybe just mention https://book.kubebuilder.io/reference/generating-crd.html#validation

Reply:
added a note for how to fetch the schema and a reference to that link